File: Week1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
int get_change(void);
int printcoins(int n);
int main()
{

    int c = get_change();
    int p = printcoins(c);
    printf("Coins: %d\n", p);
}

int get_change(void)
{
    int n;
    do
    {
        n = get_int("Enter Change: ");
    }
    while (n < 0);
    return n;
}
int printcoins(int n)
{

    int coins = 0;
    if (n == 0)
    {
        return coins;
    }
    else
    {

        coins += n / 25;
        n %= 25;

        coins += n / 10;
        n %= 10;

        coins += n / 5;
        n %= 5;

        coins += n / 1;
        n %= 1;
        return coins;
    }
}

File: Week6/sentimental-cash/cash.py
from cs50 import get_float

def print_coins(change):
    coins = 0
    if change == 0:
        return coins

    coins += change//25
    change %= 25

    coins += change//10
    change %= 10

    coins += change//5
    change %= 5

    coins += change//1
    change %= 1

    return coins

while True:
    try:
        change = get_float("Change: ")
        if change >= 0:
            break
        print("Invalid value\nEnter a non-negative value")
    except ValueError:
        print("Invalid value\nEnter a non-negative value")

change = round(change*100)
print(print_coins(change))
